consensus/propagation/commitment.go

```diff
@@ -203,8 +185,30 @@ func (blockProp *Reactor) handleCompactBlock(cb *proptypes.CompactBlock, peer p2
 		select {
 		case <-blockProp.ctx.Done():
 			return
-		case blockProp.proposalChan <- cb.Proposal:
+		case blockProp.proposalChan <- ProposalAndFrom{
```

Comment:
if this is not the proposer, we send the received proposal to the consensus reactor so that it's added to the peer state and the consensus proposal's catchup work as needed